### DISCUSSION TITLE
Additional Document Actions

### DISCUSSION THREAD

**Discussion by raulsaavedrax** [sanity-io/sanity#3835] [Sanity Studio Support] [answered]
Hello everyone!
I want to migrate a custom document action from V2 to V3. I followed this [guide](https://beta.sanity.io/docs/platform/studio/v2-to-v3/document-actions-badges).

However, I don't want to replace the Publish Action, but instead add an additional one named Preview Article, so it has the following behavior:

![image](https://user-images.githubusercontent.com/98895099/199577028-62827509-5989-482d-8b69-d89a0e91bf89.png)

In V3, instead of adding the additional action to the dropdown, it replaces the Publish Action, when I would like the publish one to be the main one.

![image](https://user-images.githubusercontent.com/98895099/199577241-506530b7-190f-4ea6-9ea9-f9e26e4fae4d.png)

Here is how I moved my v2 code to the new v3 code.

V2
```
import defaultResolve from 'part:@sanity/base/document-actions';

import { PreviewBlogArticleAction } from './PreviewBlogArticleAction';

export default function resolveDocumentActions(props) {
  return [...defaultResolve(props), PreviewBlogArticleAction];
}
```
V3

```
// inside defineConfig({})
document: {
  actions: (prev) => {
    return [...prev, PreviewBlogArticleAction];
  },
},
```

I also tried to create a custom Publish Action following the guide above along my new action to work around this, but I see that PublishAction is not exported anymore as of the new release candidate version. ([sanityv3.0.0-rc.0](https://github.com/sanity-io/sanity/releases/tag/v3.0.0-rc.0))
![image](https://user-images.githubusercontent.com/98895099/199578137-7ba8dc1c-945a-43d7-99f2-9859d518555d.png)
![image](https://user-images.githubusercontent.com/98895099/199580397-ab1b2f73-a29e-4523-9618-8a55905fa1ea.png)

 
What I am doing wrong? or is the new actions API only meant to replace predefined actions?

**Comment by geball** [comment, marked as answer]
Hi @raulsaavedrax. Please take a look at [this reply](https://github.com/sanity-io/sanity/discussions/3796#discussioncomment-4091723), which gives some context and also offers a workaround to access the array of document actions. In your case, you would want to modify the map so as not to filter out the default Publish action, but the approach should be similar.

**Reply by raulsaavedrax** [reply]
This fixed it for me 💯 Thank you! @geball
